packages/@aws-cdk/aws-batch/lib/job-queue.ts

```diff
@@ -0,0 +1,139 @@
+
+  /**
+   * The priority of the job queue. Job queues with a higher priority (or a higher integer value for the priority parameter) are evaluated first
+   * when associated with the same compute environment. Priority is determined in descending order, for example, a job queue with a priority value
+   * of 10 is given scheduling preference over a job queue with a priority value of 1.
+   *
+   * @default 1
+   */
+  readonly priority?: number;
+
+  /**
+   * The state of the job queue. If set to true, it is able to accept jobs.
+   *
+   * @default true
+   */
+  readonly enabled?: boolean;
+}
+
+/**
+ * Properties of a Job Queue
+ */
+export interface IJobQueue extends IResource {
+  /**
+   * The ARN of this batch job queue
+   *
+   * @attribute
+   */
+  readonly jobQueueArn: string;
+
+  /**
+   * A name for the job queue.
+   *
+   * Up to 128 letters (uppercase and lowercase), numbers, hyphens, and underscores are allowed.
+   *
+   * @attribute
+   */
+  readonly jobQueueName: string;
+}
+
+/**
+ * Batch Job Queue
+ *
+ * Defines a batch job queue to define how submitted batch jobs
+ * should be ran based on specified batch compute environments.
+ */
+export class JobQueue extends Resource implements IJobQueue {
+  /**
+   * Fetches an existing batch job queue by its amazon resource name.
+   *
+   * @param scope
+   * @param id
+   * @param jobQueueArn
+   */
+  public static fromJobQueueArn(scope: Construct, id: string, jobQueueArn: string): IJobQueue {
+    const stack = Stack.of(scope);
+    const jobQueueName = stack.parseArn(jobQueueArn).resource;
+
+    class Import extends Resource implements IJobQueue {
+      public readonly jobQueueArn = jobQueueArn;
+      public readonly jobQueueName = jobQueueName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly jobQueueArn: string;
+  public readonly jobQueueName: string;
+
+  constructor(scope: Construct, id: string, props: JobQueueProps = {}) {
+    super(scope, id, {
+      physicalName: props.jobQueueName,
+    });
+
+    const jobQueue = new CfnJobQueue(this, 'Resource', {
+      computeEnvironmentOrder: props.computeEnvironmentOrder
+      ? props.computeEnvironmentOrder.map(cp => ({
+        computeEnvironment: cp.computeEnvironment.computeEnvironmentArn,
+        order: cp.order,
+      } as CfnJobQueue.ComputeEnvironmentOrderProperty))
+      : [
+        {
+          // Get an AWS Managed Compute Environment
+          computeEnvironment: new ComputeEnvironment(this, 'Resource-Batch-Compute-Environment').computeEnvironmentArn,
+          order: 1,
+        },
+      ],
+      jobQueueName: this.physicalName,
+      priority: props.priority || 1,
+      state: props.enabled === undefined ? 'ENABLED' : (props.enabled ? 'ENABLED' : 'DISABLED'),
+    });
+
+    this.jobQueueArn = this.getResourceArnAttribute(jobQueue.ref, {
+      service: 'batch',
+      resource: 'job-queue',
+      resourceName: this.physicalName,
```

Comment:
Again a discrepancy between the `from` method, and this ARN construction.

Reply:
I am unsure how I should change this or so. Could you provide an example?

Reply:
Same as my comment here: https://github.com/aws/aws-cdk/pull/4732#discussion_r363457612